tooling/tracer/src/lib.rs

```diff
@@ -132,11 +146,12 @@ pub fn trace_circuit<B: BlackBoxFunctionSolver<FieldElement>>(
     );
 
     if tracing_context.debug_context.get_current_opcode_location().is_none() {
-        println!("Warning: circuit contains no opcodes; generating empty trace");
-        return Ok(TraceArtifact::new());
+        println!("Warning: circuit contains no opcodes; generating no trace");
+        return Ok(());
     }
 
     let mut steps = 0;
+    let mut is_tracer_started = false;
```

Comment:
can we explicitly call tracer `start`here: it doesn't actually register a step, but the toplevel call(context): this would simplify the update_record

Reply:
We need to know the line and file at which to start, which is only available after the first debugging step.

Or am I missing something? Do you mean calling it without arguments or something like this?

Reply:
It wasn't clear from your reply what would be useful, but I'm assuming "calling start without arguments".

Reply:
Better?